packages/search-in-workspace/src/browser/components/search-in-workspace-textarea.tsx

```diff
@@ -0,0 +1,178 @@
+// *****************************************************************************
+// Copyright (C) 2021 Ericsson and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import * as React from '@theia/core/shared/react';
+import { Key, KeyCode } from '@theia/core/lib/browser';
+import debounce = require('@theia/core/shared/lodash.debounce');
+
+interface HistoryState {
+    history: string[];
+    index: number;
+};
+type TextareaAttributes = React.TextareaHTMLAttributes<HTMLTextAreaElement>;
+
+export class SearchInWorkspaceTextArea extends React.Component<TextareaAttributes, HistoryState> {
+    static LIMIT = 100;
+    static MAX_ROWS = 7;
+
+    private textarea = React.createRef<HTMLTextAreaElement>();
+
+    constructor(props: TextareaAttributes) {
+        super(props);
+        this.state = {
+            history: [],
+            index: 0,
+        };
+    }
+
+    override componentDidMount(): void {
+        this.resizeTextarea();
+    }
+
+    override componentDidUpdate(prevProps: Readonly<TextareaAttributes>, prevState: Readonly<HistoryState>): void {
+        this.resizeTextarea();
```

Comment:
Theia already uses a lib for a resizable `textarea` at the SCM commit comment. Can it not be reused instead of reimplementing the resizing? Do you want to achieve something else?

https://github.com/eclipse-theia/theia/blob/109adc41f1340ef2e5aee862ece1cd49ca792323/packages/scm/package.json#L12

Reply:
I originally overlooked that there was already a dependency to react-autosize-textarea and implemented my own solution. When I found out about the lib, I tried to adopt it but ran into some problems, so I kept my own code.

I now re-evaluted using react-autosize-textarea and was able to make it work with the following two fixes necessary:
- I had to compute the maximum height myself because the maxRows property had no effect
- I had to filter out the onResize property due to an incompatibility in types

Changes are pushed.